Fixed Proposal Details Page, Create CDP Button and CDP List

MonikaCat opened the pull request:
<!-- < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < ☺
v                               ✰  Thanks for creating a PR! ✰    
v    Before smashing the submit button please review the checkboxes.
v    If a checkbox is n/a - please still include it but + a little note why
☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  -->

## Description
<!-- Small description. What have you fixed or changed? -->
Fixes #438 

## Checklist
- [x] Targeted PR against correct branch.
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Run `meteor npm run lint`
- [x] Added an entry to the `CHANGELOG.md` file.
- [x] Re-reviewed `Files changed` in the Github PR explorer.

kwunyeung reviewed (changes requested):
It shows NaN and undefined when there is no available `Hard` in the account.

![image](https://user-images.githubusercontent.com/932636/103277569-51f0eb00-4a04-11eb-9e13-8a8a8f4778d1.png)

I have claimed some vesting Hard in our account and the detail is not being shown correctly.

![image](https://user-images.githubusercontent.com/932636/103277618-6e8d2300-4a04-11eb-954f-64284e23ca90.png)

This is the account address `kava14kn0kk33szpwus9nh8n87fjel8djx0y08wynpx`. Please see if the structure with the vesting token breaks the page.

kwunyeung commented:
![](https://github.trello.services/images/mini-trello-icon.png) [CDP list display and Ledger button fix](https://trello.com/c/kvIa3fBP/6-cdp-list-display-and-ledger-button-fix)

kwunyeung reviewed (commented):
Does this means there are different CDPs from different denoms? Should we add a selector for user to choose which collateral they want to explore?

kwunyeung commented:
I have just done a query `kvcli q cdp cdps --collateral-type hard-a` and there are a lot of CDPs.

MonikaCat commented:
> I have just done a query `kvcli q cdp cdps --collateral-type hard-a` and there are a lot of CDPs.

Ok I see! Looks like the `cdp/cdps` query tracks only CDPs created with BNB. I've changed that and added Tab selection for each collateral type to allow users explore the collateral they're interested in.

kwunyeung commented on imports/ui/accounts/Account.jsx:
```diff
@@ -695,9 +739,14 @@ export default class AccountDetails extends Component {
-                                        <Col xs={12} className="value-3 text-right">{this.findCoin(this.state.total, coin3)}</Col>
-                                        <Col xs={12} className="dollar-value-3 text-right text-secondary">~{numbro((this.findValue(this.state.total, coin3)) * this.state.usdxPrice).format("$0,0.0000a")} ({numbro(this.state.usdxPrice).format("$0,0.00")}/{Meteor.settings.public.coins[5].displayName})</Col>
+                                        <Col xs={12} className="value-3 text-right">{this.findCoin(this.state.total, coin4)}</Col>
+                                        <Col xs={12} className="dollar-value-3 text-right text-secondary">~{numbro((this.findValue(this.state.total, coin4)) * this.state.hardPrice).format("$0,0.0000a")} ({numbro(this.state.hardPrice).format("$0,0.00")}/{Meteor.settings.public.coins[8].displayName})</Col>
```
I can't test the account operations as I don't know how your `settings.json` looks like right now. Please provide information on the coin changes in the setting as it's all hard coded right now.

MonikaCat replied:
Ok! I will send you my settings.json file now

kwunyeung commented on imports/ui/cdp/CDPs.jsx:
```diff
@@ -33,9 +60,89 @@ export default class CDPs extends Component{
+                </TabPane>
+                <TabPane tabId="cdp-hard">
+                    <Switch>
+                        <Route exact path="/cdps" component={this.CDPList} />
```
What's the usage of creating multiple routes with the same route here?

MonikaCat replied:
Ohh! Fixed that!

kwunyeung commented on imports/ui/cdp/CDPs.jsx:
```diff
@@ -5,22 +5,49 @@ import List from '/imports/ui/cdp/List.jsx';
+
+    toggleTab = (tab, newCollateralType) => {
+        if (this.state.cdpActiveTab !== tab) {
+            this.setState({
```
If the tabs are all the same, how about not switching tab here and only use one `TabPane` then use pass the correct prop to the `CDPList` component to get a new lis? I see you have `props` in `CDPList` but not passing any props into to it.

MonikaCat replied:
Ok updated!

kwunyeung reviewed (commented):
The code LGTM. I'm going to test now.

kwunyeung commented:
2 issues.

1. I think the CDP calculations with HARD is wrong. The usd value is not correct.
![image](https://user-images.githubusercontent.com/932636/103991356-80718180-51cd-11eb-9f8e-9e82fbe89eed.png)

2. When switching different denom in CDP, it's better to clear the list and show a loading indicator then wait for new data to feed in. Currently, when you switch the denom in different tabs, it keeps waiting but the existing data keeps there and it's strange. For example, when you are reading the CDP list of Kava and then you switch BNB, the list wouldn't be refreshed until the new data have arrived. In that case the tab has been switched to BNB but the CDP list is still showing the CDP of Kava.
